File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/interpreter/lab/UnaryPlus.java
// UnaryPlus.java

import java.io.PrintWriter;

public class UnaryPlus extends Unary
{
	public UnaryPlus(Expression aOperand1)
	{
		super(aOperand1);
	}

	public void output(PrintWriter aPrintWriter)
	{
		outputPrefix(aPrintWriter, "+");
	}
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/interpreter/lab/unary_plus.cpp
// unary_plus.cpp

#include <iostream>
#include "dp_exception.h"
class expression;
#include "unary.h"
#include "unary_plus.h"

unary_plus::unary_plus(expression * a_operand1)
: unary(a_operand1)
{
}

void unary_plus::output(std::ostream & a_ostream) const
{
	output_prefix(a_ostream, "+");
}
